extensions/elasticsearch-rest-client/runtime/src/main/java/io/quarkus/elasticsearch/restclient/lowlevel/runtime/ElasticsearchConfig.java

```diff
@@ -0,0 +1,83 @@
+package io.quarkus.elasticsearch.restclient.lowlevel.runtime;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(phase = ConfigPhase.RUN_TIME)
+public class ElasticsearchConfig {
+
+    /**
+     * The list of hosts of the Elasticsearch servers.
+     */
+    @ConfigItem(defaultValue = "localhost:9200")
+    public List<String> hosts;
+
+    /**
+     * The protocol to use when contacting Elasticsearch servers.
+     * Set to "https" to enable SSL/TLS.
+     */
+    @ConfigItem(defaultValue = "http")
+    public String protocol;
+
+    /**
+     * The username used for authentication.
+     */
+    @ConfigItem
+    public Optional<String> username;
+
+    /**
+     * The password used for authentication.
+     */
+    @ConfigItem
+    public Optional<String> password;
+
+    /**
+     * The connection timeout.
+     */
+    @ConfigItem(defaultValue = "1S")
+    public Duration connectionTimeout;
+
+    /**
+     * The socket timeout.
+     */
+    @ConfigItem(defaultValue = "30S")
+    public Duration socketTimeout;
+
+    /**
+     * The number of IO thread.
+     * By default, this is the number of locally detected processors.
+     *
+     * @see <a href="https://www.elastic.co/guide/en/elasticsearch/client/java-rest/current/_number_of_threads.html">number of
+     *      threads</a>
+     */
+    @ConfigItem
+    public Optional<Integer> ioThreadCounts;
+
```

Comment:
Connections settings, which I are quite important, are missing:

* `max_connections`: the maximum number of connections to the cluster.
* `max_connections_per_route`: the maximum number of connections to each node in the cluster.

See https://github.com/hibernate/hibernate-search/blob/9da99d7de8f346aa0ec81ae2ce0a1fa2c1657b2a/backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/client/impl/ElasticsearchClientFactoryImpl.java#L201-L200

Reply:
The documentation now only references this one. See https://www.elastic.co/guide/en/elasticsearch/client/java-rest/current/_number_of_threads.html

Reply:
I think it's because of https://www.elastic.co/guide/en/elasticsearch/client/java-rest/current/_others.html . They basically send you to the Apache HTTP client doc.

I would trust Yoann on that one :).

Reply:
I'm not sure I'm following. The settings I mentioned are settings of the underlying HTTP client, not of the Elasticsearch Rest client. So indeed, they don't have dedicated entries in the Elasticsearch Rest client documentation. They are still relevant, though. See https://www.elastic.co/guide/en/elasticsearch/client/java-rest/current/_others.html

Reply:
Yeah, I understand it now, it's different things ;)